bokeh/charts/horizon.py

```diff
@@ -0,0 +1,230 @@
+                (default: False)
+            facet (bool, optional): generate multiple areas on multiple
+                separate charts for each series if True. Defaults to
+                False
+            xgrid (bool, optional): whether to display x grid lines
+                (default: True)
+            ygrid (bool, optional): whether to display y grid lines
+                (default: True)
+            pos_color (hex color string): the color of the positive folds
+                (default: #006400)
+            neg_color (hex color string): the color of the negative folds
+                (default: #6495ed)
+
+        Attributes:
+            source (obj): datasource object for your plot,
+                initialized as a dummy None.
+            xdr (obj): x-associated datarange object for you plot,
+                initialized as a dummy None.
+            ydr (obj): y-associated datarange object for you plot,
+                initialized as a dummy None.
+            groups (list): to be filled with the incoming groups of data.
+                Useful for legend construction.
+            data (dict): to be filled with the incoming data and be passed
+                to the ColumnDataSource in each chart inherited class.
+                Needed for _set_And_get method.
+            attr (list): to be filled with the new attributes created after
+                loading the data dict.
+                Needed for _set_And_get method.
+        """
+        self.values = values
+        self.source = None
+        self.xdr = None
+        self.ydr = None
+        self.groups = []
+        self.series = []
+        self.fold_height = {}
+        self.max_y = 0
+        self.data = dict()
+        self.attr = []
+        self.index = index
+        self.nb_folds = nb_folds
+        self.pos_color = pos_color
+        self.neg_color = neg_color
+
+        super(Horizon, self).__init__(
+            title, xlabel, ylabel, legend, xscale, yscale, width, height,
+            tools, filename, server, notebook, facet, xgrid, ygrid
+        )
+
+    def get_data(self):
+        """Take the x/y data from the horizon values.
+
+        It calculates the chart properties accordingly. Then build a dict
+        containing references to all the points to be used by
+        the multiple area glyphes inside the ``draw`` method.
+
+        """
+
+        def fold_coordinates(y, fold_no, fold_height, y_origin=0):
+            """ Function that calculate the coordinates for a value given a fold
+            """
+            height = fold_no * fold_height
+            quotient, remainder = divmod(abs(y), float(height))
+            v = fold_height
+            if math.floor(quotient) == 0:
+                v = (remainder - height + fold_height) if remainder >= height - fold_height else 0
+            return (v + y_origin, fold_height + y_origin) if y > 0 else (y_origin, fold_height - v + y_origin)
```

Comment:
Why currying here? I can't get any advantage in this case. Would just define it outside...